libs/api/src/lib/challenges/cfTurnstile.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  Challenge,
+  ChallengeProvider,
+  ChallengeValidationProps,
+  ChallengeValidationReturn
+} from './challengeProvider'
+
+export class CfTurnstile implements ChallengeProvider {
+  constructor(private turnstile_secret_key: string) {}
+
+  async generateChallenge(testing = false): Promise<Challenge> {
+    return {
+      challengeID: 'not supported is turnstile!',
```

Comment:
Should probably not be set.

Reply:
@Itrulia  What do you mean by "should not be set"?

Reply:
The challengeID should not be set, if it's not supported.

Reply:
Is this possible on graphql level if it is required for algebraic but not for turnstile captcha?

Reply:
Yes, by using Unions (like Blocks). Which would then also require the type like I wrote in another comment. The frontend needs the type to know how to display it.